azure-pipelines.yml

```diff
@@ -0,0 +1,56 @@
+trigger:
+- main
+
+pr: 
+- main
+
+resources:
+  containers:
+  - container: ubuntu_20_04_ros2_foxy
+    image: ros:foxy-ros-base
+    options: '-u root:sudo'
+  - container: ubuntu_22_04_ros2_humble
+    image: ros:humble-ros-base
+    options: '-u root:sudo'
+  - container: ubuntu_22_04_ros2_rolling
+    image: ros:rolling-ros-base
+    options: '-u root:sudo'
+
+
+strategy:
+  matrix:
+    ubuntu_20_04_foxy:
+      buildContainer: ubuntu_20_04_ros2_foxy
+    ubuntu_22_04_humble:
+      buildContainer: ubuntu_22_04_ros2_humble
+    ubuntu_22_04_rolling:
+      buildContainer: ubuntu_22_04_ros2_rolling
+
+container: $[ variables ['buildContainer'] ]
+
+steps:
+  - script: chmod 777 CI/dependencies_setup.sh && ./CI/dependencies_setup.sh
```

Comment:
Ok, then it can remain the chmod command before running. 
Wouldn't be better to make all scripts files executable directly in repo? Something like 
`git add --chmod=+x -- file_name` then push it.

Reply:
I changed the permissions and triggered the job again. It passed.